### DISCUSSION TITLE
Increasing --max-old-space-size for the extension node process

### DISCUSSION THREAD

**Discussion by shashankvidiyalaC3** [microsoft/vscode-discussions#867] [Extension Development QnA] [answered]
Since VsCode activates the extensions we don't have control over node options  is there any way through vscode API to increase the memory limit for the extension process

**Comment by tamuratak** [comment, marked as answer]
No. See

https://github.com/microsoft/vscode/issues/127105#issuecomment-1124504462

> the heap limit is due to the pointer compression feature from v8 https://v8.dev/blog/v8-release-92#shared-pointer-compression-cage and we cannot disable the limit at runtime, it has to be done by disabling this feature at build time. Disabling pointer compression is not an option, since it will deviate our runtime from upstream electron and also get rid of the memory savings, sandbox features that chromium is working on.

A workaround the built-in TypeScript extension has adopted is launching the server with a vanilla Node.js if available.

- https://github.com/microsoft/vscode/pull/191019

**Reply by shashankvidiyalaC3** [reply]
Thanks for explaining
